doc/i18n.md

```diff
@@ -0,0 +1,71 @@
+
+Cockpit Internationalization
+============================
+
+All human-visible strings on Cockpit's web pages should be translatable. This
+document describes the entire internationalization process, which uses [GNU gettext](https://www.gnu.org/software/gettext/) and related tools as much as possible.
+
+Marking strings
+---------------
+Strings that are visible to humans and which should be translated must be
+marked as such on Cockpit pages, in one of two ways:
+
+ * If the string appears in a HTML source, it must be enclosed in a tag with
+   the `translatable="yes"` attribute. Example:
+
+   ```html
+   <span translatable="yes">Image</span>
+   ```
+
+ * If the string appears in JavaScript source, it must use the
+   [cockpit.gettext()](http://cockpit-project.org/guide/latest/cockpit-locale.html#cockpit-locale-gettext)
+   or a related function, called through an alias `_()`, with double-quoted
+   strings, so that `xgettext` can recognize it. Example:
+
+   ```js
+   const _ = cockpit.gettext
+
+   var translated = _("Time")
+
+   // in a JSX block:
+   <tr>
+     <th>{ _("Image") }</th>
+   </tr>
+   ```
```

Comment:
suggestion: the `cockpit.format()` function can be pointed out as well for parameterized messages.

Reply:
It's not directly related to translations, but indeed useful in conjunction with it, so I added a reference.